src/modules/wallet/dto/csv-item.dto.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  IsString,
+  IsOptional,
+  IsNotEmpty,
+  IsNumber,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from 'class-validator';
+
+// Custom validator to check for unique `wallet_name` values in the csvJson array
+@ValidatorConstraint({ async: false })
+export class UniqueWalletNameConstraint
+  implements ValidatorConstraintInterface
+{
+  validate(csvJson: CsvItemDto[]): boolean {
+    const walletNames = csvJson.map((item) => item.wallet_name);
+    return walletNames.length === new Set(walletNames).size;
+  }
+
+  defaultMessage(): string {
+    return 'Each wallet_name in csvJson must be unique.';
+  }
+}
+
+export class CsvItemDto {
```

Comment:
@Kpoke i've added this `CsvItemDto` for validation purposes for both `batchCreateWallet` and `batchTransferWallet`. let me know if this is what you're looking for. thanks!!